lib/record_store/provider/cloudflare.rb

```diff
@@ -36,44 +36,46 @@ def zones
         end
       end
 
-      private
-
-      def add(record, zone)
-        zone_id = zone_name_to_id(zone)
-
-        api_body = build_api_body(record)
-
-        retry_on_connection_errors do
-          client.post("/client/v4/zones/#{zone_id}/dns_records", api_body)
+      def apply_changeset(changeset, stdout = $stdout)
+        deletes = []
+        patches = []
+        posts = []
+        puts = []
+
+        changeset.changes.each do |change|
+          case change.type
+          when :removal
+            stdout.puts "Removing #{change.record}..."
+            deletes << { id: change.record.id }
+          when :addition
+            stdout.puts "Creating #{change.record}..."
+            posts << build_api_body(change.record)
+          when :update
+            stdout.puts "Updating record with ID #{change.id} to #{change.record}..."
+            patches << build_api_body(change.record).merge(id: change.id)
```

Comment:
is there a reason you're using patches instead of puts?
in this case am I right that either could be used since we're likely providing all fields anyways?